src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -572,7 +572,7 @@ bool ObjectMonitor::deflate_monitor() {
 
     // Make a zero contentions field negative to force any contending threads
     // to retry. This is the second part of the async deflation dance.
-    if (Atomic::cmpxchg(&_contentions, (jint)0, -max_jint) != 0) {
+    if (Atomic::cmpxchg(&_contentions, (int)0, INT32_MIN) != 0) {
```

Comment:
INT_MIN

Do we really need the cast on 0?

Reply:
The original cast of `(jint)0` was to get the right `cmpxchg()` selected.
I _think_ that with the switch to INT32_MIN, a regular `0` instead of `(int)0`
should compile.

Reply:
Yes, changing to INT_MIN is more correct and I don't need the cast to int now for zero.  Recompiling on all platforms, including the picky ones for verification.  Thanks!